dev/com.ibm.ws.session.cache/src/com/ibm/ws/session/store/cache/CacheHashMap.java

```diff
@@ -1165,30 +1207,36 @@ protected void removePersistedSession(String id) {
 
         //If the app calls invalidate, it may not be removed from the local cache yet.
         superRemove(id);
+        try {
+            if (trace && tc.isDebugEnabled())
+                tcInvoke(tcSessionMetaCache, "getAndRemove", id);
 
-        if (trace && tc.isDebugEnabled())
-            tcInvoke(tcSessionMetaCache, "getAndRemove", id);
-
-        ArrayList<?> removed = sessionMetaCache.getAndRemove(id);
+            ArrayList<?> removed = null;
```

Comment:
Why was this split into two assignments?  A single assignment was more efficient.